sdcm/utils.py

```diff
@@ -239,6 +252,87 @@ def list_logs_by_test_id(test_id):
+    monitor_stack_archives = []
+    for f in stored_files_by_test_id:
+        if f['type'] in ['monitoring_data_stack', 'prometheus']:
+            monitor_stack_archives.append(f)
+
+    if not monitor_stack_archives and len(monitor_stack_archives) < 2:
+        logger.warning('There is no available archive files for monitoring data stack restoring for test id : {}'.format(test_id))
+        return False
+
+    for arch in monitor_stack_archives:
+        logger.info('Download file {} to directory {}'.format(arch['link'], monitor_stack_base_dir))
+        local_path_monitor_stack = S3Storage().download_file(arch['link'], dst_dir=monitor_stack_base_dir)
+        monitor_stack_workdir = os.path.dirname(local_path_monitor_stack)
+        monitoring_stack_archive_file = os.path.basename(local_path_monitor_stack)
+        logger.info('Extracting data from archive {}'.format(arch['file_path']))
+        if arch['type'] == 'prometheus':
+            monitor_stack_data_dir = os.path.join(monitor_stack_workdir, 'monitor_data_dir')
+            cmd = dedent("""
+                mkdir -p {data_dir}
+                cd {data_dir}
+                cp ../{archive} ./
+                tar -xvf {archive}
+                chmod -R 777 {data_dir}
+                """.format(data_dir=monitor_stack_data_dir,
+                           archive=monitoring_stack_archive_file))
+            result = lr.run(cmd, ignore_status=True)
+        else:
+            branches = re.search('(?P<monitoring_branch>branch-[0-9]+\.[0-9]+?)_(?P<scylla_version>[\d]\.[\d]+?)',
+                                 monitoring_stack_archive_file)
+            monitoring_branch = branches.group('monitoring_branch')
+            scylla_version = branches.group('scylla_version')
+            cmd = dedent("""
+                cd {workdir}
+                tar -xvf {archive}
+                """.format(workdir=monitor_stack_workdir, archive=os.path.basename(local_path_monitor_stack)))
+            result = lr.run(cmd, ignore_status=True)
+        if not result.ok:
+                logger.warning("During restoring file {} next errors occured:\n {}".format(arch['link'], result))
+                return False
+        logger.info("Extract data finished")
+
+    logger.info('Monitoring stack files available {}'.format(monitor_stack_workdir))
+
+    monitoring_dockers_dir = os.path.join(monitor_stack_workdir, 'scylla-grafana-monitoring-{}'.format(monitoring_branch))
+
+    for i in range(5):
+        logger.info('Start Attempt {}'.format(i + 1))
+        lr.run('cd {}; ./kill-all.sh'.format(monitoring_dockers_dir))
+        cmd = dedent("""cd {monitoring_stack_dir};
+                ./start-all.sh \
+                -s {monitoring_stack_dir}/config/scylla_servers.yml \
+                -n {monitoring_stack_dir}/config/node_exporter_servers.yml \
+                -d {monitoring_data_dir} -v {scylla_version}""".format(monitoring_stack_dir=monitoring_dockers_dir,
```

Comment:
use the same variable in formatted string and format with **locals()

Reply:
fixed